app/client/src/widgets/TableWidgetV2/component/VirtualTable/useRowHeightMeasurement.tsx

```diff
@@ -0,0 +1,133 @@
+import { useEffect, type RefObject } from "react";
+import type { Row as ReactTableRowType } from "react-table";
+import type { VariableSizeList } from "react-window";
+import { ColumnTypes } from "widgets/TableWidgetV2/constants";
+import { TABLE_SIZES } from "../Constants";
+
+interface CellWithColumnProps {
+  column: {
+    columnProperties?: {
+      allowCellWrapping?: boolean;
+      columnType: ColumnTypes;
+    };
+  };
+}
+
+interface UseRowHeightMeasurementProps {
+  index: number;
+  row: ReactTableRowType<Record<string, unknown>>;
+  rowRef?: React.RefObject<HTMLDivElement>;
+  rowHeights?: RefObject<{ [key: number]: number }>;
+  rowNeedsMeasurement?: RefObject<{ [key: number]: boolean }>;
+  listRef?: RefObject<VariableSizeList> | null;
+  forceUpdate: number;
+  isInfiniteScrollEnabled?: boolean;
+}
+
+export function useRowHeightMeasurement({
+  forceUpdate,
+  index,
+  isInfiniteScrollEnabled,
+  listRef,
+  row,
+  rowHeights,
+  rowNeedsMeasurement,
+  rowRef,
+}: UseRowHeightMeasurementProps) {
+  useEffect(() => {
+    if (
+      !isInfiniteScrollEnabled ||
+      !rowNeedsMeasurement ||
+      !rowHeights ||
+      !rowRef
+    )
+      return;
+
+    if (
+      rowNeedsMeasurement.current &&
+      rowNeedsMeasurement.current[index] === false
+    ) {
+      return;
+    }
+
+    const element = rowRef.current;
+
+    if (!element || !row) return;
```

Comment:
Why do we have multiple ifs? you can improve the readability by using variable and single if: 
```suggestion
const hasPreRequisitesForInfiniteScroll  = isInfiniteScrollEnabled && rowHeights && rowRef;

const isAlreadyCalculated = rowNeedsMeasurement && 
      rowNeedsMeasurement.current &&
      rowNeedsMeasurement.current[index] === false;
      
const hasRowDataAndUIElement = rowRef.current && row;

    if (isAlreadyCalculated && !hasPreRequisitesForInfiniteScroll ||  !hasRowDataAndUIElement) return;
```

Reply:
Done.